syncer/syncer.go

```diff
@@ -718,6 +731,68 @@ func (s *Syncer) addCount(isFinished bool, queueBucket string, tp opType, n int6
 	}
 }
 
+func (s *Syncer) jobsIsEmpty() bool {
+	for idx := range s.jobs {
+		if len(s.jobs[idx]) > 0 {
+			return false
+		}
+	}
+	return true
+}
+
+// updateReplicationLag calculates syncer's replication lag by job, it called after every batch dml job / one skip job / one ddl
+// job is flushed to target db.
+func (s *Syncer) updateReplicationLag(job *job, queueBucketName string) {
+	// when job is nil mean no job in this bucket, need do reset this bucket lag to 0
+	if job == nil {
+		s.workerLagMap[queueBucketName].Store(0)
+		// when job is nill and s.jobs is emepty, means all event is consumed, we update lag to 0
+		if s.jobsIsEmpty() {
```

Comment:
we could check `workerLagMap` all zero instead of checking the length of channel, maybe that's faster. (you could verify it by a benchmark)

Reply:
i think check job queue is empty directly has more clean logic than check the workerLagMap ?

Reply:
channel has a lock, which is expensive than atomic operation

Reply:
you are right, fixed